### PR TITLE
Add `process_utils` with helper method to start module with sdnotify hooks

### PR DESCRIPTION
**Pull request by NeonDaniel**

### CODE REVIEW

**Comment by JarbasAl** [comment]
should this go directly into ovos_utils and ovos services made to share this logic? seems like we have this snippet duplicated all over

File: neon_utils/process_utils.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sdnotify
```
Comment (JarbasAl) [inline neon_utils/process_utils.py:29]: to keep this dependency optional can it be moved to inside the function? 

a try: except: could then be used to launch the service with kwargs only if sdnotify not available

just thinking about platforms without systemd, they are rare but exist out there. not to mention a subset of the utils package might even be useful in windows

Comment (NeonDaniel) [reply neon_utils/process_utils.py]: Makes sense, especially since sdnotify isn't a listed requirement

**Review by JarbasAl — APPROVED** [review]

**Comment by NeonDaniel** [comment]
> should this go directly into ovos_utils and ovos services made to share this logic? seems like we have this snippet duplicated all over

I was going to suggest that; I did this to avoid maintaining duplicate code in the `libexec` files I copied from ovos-buildroot. @j1nx might have some insight here; I know watchdog kwargs are also there but not yet implemented but I'm not sure if those are common to all services or unique

**Pull request merged**